llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -984,32 +984,38 @@ void State::addInfoFor(BasicBlock &BB) {
       continue;
     }
 
-    if (match(&I, m_Intrinsic<Intrinsic::ssub_with_overflow>())) {
+    auto *II = dyn_cast<IntrinsicInst>(&I);
+    Intrinsic::ID ID = II ? II->getIntrinsicID() : Intrinsic::not_intrinsic;
+    switch (ID) {
+    case Intrinsic::assume:
+      Value *A, *B;
+      CmpInst::Predicate Pred;
+      if (I.getNumOperands() >= 1 &&
```

Comment:
Is this needed? `assume` calls should only have a single argument?

Reply:
you are right,  not needed